src/coreclr/vm/multicorejitplayer.cpp

```diff
@@ -1349,53 +1106,149 @@ HRESULT MulticoreJitProfilePlayer::PlayProfile()
+            else
+            {
+                hr = HandleModuleRecord(pRec);
+            }
+        }
+        else if (rcdTyp == MULTICOREJIT_MODULEINF_RECORD_ID)
         {
-            if (rcdTyp == MULTICOREJIT_MODULE_RECORD_ID)
+            unsigned moduleIndex = data1 & (MAX_MODULES - 1);
+            unsigned level = (data1 & ((MAX_MODULE_LEVELS - 1) << MODULE_LEVEL_OFFSET)) >> MODULE_LEVEL_OFFSET;
+
+            hr = HandleModuleInfoRecord(moduleIndex, level);
+        }
+        else if (rcdTyp == MULTICOREJIT_METHODINF_RECORD_ID)
+        {
+            // Find all subsequent methods and jit/load them reversed if reversed order is required
+            bool reversedOrder = true;
+
+            if (reversedOrder)
             {
-                const ModuleRecord * pRec = (const ModuleRecord * ) pBuffer;
+                bool isMethod = true;
+                const BYTE * pCurBuf = pBuffer;
+                unsigned curSize = nSize;
+
+                unsigned sizes[MAX_WALKBACK] = {0};
+                int count = 0;
+
+                do
+                {
+                    unsigned curdata2 = * (((const unsigned *) pCurBuf) + 1);
+                    unsigned currcdLen = curdata2 >> SIGNATURE_LENGTH_OFFSET;
+
+                    if (currcdLen > curSize)
+                    {
+                        hr = COR_E_BADIMAGEFORMAT;
+                        break;
+                    }
+
+                    sizes[count] = currcdLen;
+                    count++;
+
+                    pCurBuf += currcdLen;
+                    curSize -= currcdLen;
+
+                    if (curSize == 0)
+                    {
+                        break;
+                    }
 
-                if (((unsigned)(pRec->lenModuleName
-                    + pRec->lenAssemblyName
-                    ) > (rcdLen - sizeof(ModuleRecord))) ||
-                    (m_moduleCount >= m_headerModuleCount))
+                    unsigned curdata1 = * (const unsigned *) pCurBuf;
+                    isMethod = RecorderInfo::isMethod(curdata1);
+                }
+                while (isMethod && count < MAX_WALKBACK);
+
+                _ASSERTE(count > 0);
+                if (count > 1)
                 {
-                    hr = COR_E_BADIMAGEFORMAT;
+                    MulticoreJitTrace(("Jit backwards %d methods",  count));
                 }
-                else
+
+                int i = count - 1;
+                for (; (SUCCEEDED(hr)) && i >= 0; --i)
                 {
-                    hr = HandleModuleRecord(pRec);
+                    pCurBuf -= sizes[i];
+
+                    unsigned curdata1 = * (const unsigned *) pCurBuf;
+                    unsigned curmoduleIndex = curdata1 & (MAX_MODULES - 1);
+                    unsigned curflags = curdata1 & METHOD_FLAGS_MASK;
+
+                    unsigned curdata2 = * (((const unsigned *) pCurBuf) + 1);
+                    unsigned cursignatureLength = curdata2 & MAX_SIGNATURE_LENGTH;
```

Comment:
I think this should either be `& (MAX_SIGNATURE_LENGTH - 1)` (if its value is `0x10000`), or `& MAX_SIGNATURE_LENGTH` (if its value is changed to `0xffff` as I suggested in another comment). In any case it appears the length is not used, so it could just be removed.

Reply:
Thanks for noticing! I've fixed this typo, but forgot to update it here